drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-backend/src/main/java/org/drools/workbench/screens/guided/dtable/backend/server/conversion/BRLColumnSubHeaderBuilderDefault.java

```diff
@@ -0,0 +1,169 @@
+        dt.getData().add(list);
+
+        subHeaderBuilder.makeHeaderAndTitle(SubHeaderBuilder.CONDITION,
+                                            brlColumn.getHeader());
+        subHeaderBuilder.getFieldRow().createCell(subHeaderBuilder.getTargetColumnIndex()).setCellValue(replaceTempVars(brlColumn.getChildColumns(),
+                                                                                                                        subString(GuidedDTDRLPersistence.getInstance().marshal(dt), "when", "then")));
+    }
+
+    private List<DTCellValue52> setUpVarNamesWithTemps(final List<? extends BRLVariableColumn> childColumns) {
+        final List<DTCellValue52> result = new ArrayList<>();
+
+        for (int i = 0; i < childColumns.size(); i++) {
+            final BRLVariableColumn brlConditionVariableColumn = childColumns.get(i);
+            if (brlConditionVariableColumn.getVarName().isEmpty()) {
+                result.add(new DTCellValue52(true));
+            } else {
+                String varName = brlConditionVariableColumn.getVarName();
+                String key = "";
+                if (!varName.startsWith("$")) {
+                    key = varName;
+                    varName = "@{" + varName + "}";
+                } else {
+                    key = Integer.toString(i);
+                    varName = "@{" + i + "}";
+                }
+                varListInOrder.put(brlConditionVariableColumn.getVarName(), key);
+                result.add(new DTCellValue52(varName));
+            }
+        }
+
+        return result;
+    }
+
+    private String replaceTempVars(final List<? extends BRLVariableColumn> childColumns,
+                                   final String marshal) {
+        int varIndex = 1;
+        String result = marshal;
+        for (BRLVariableColumn childColumn : childColumns) {
+
+            final String var = varListInOrder.get(childColumn.getVarName());
+
+            final Pattern pattern = Pattern.compile("@\\{(" + var + ")}");
+            final Matcher matcher = pattern.matcher(result);
+            result = matcher.replaceAll("\\$" + varIndex++);
+        }
+        return result;
+    }
+
+    private String subString(final String marshal,
+                             final String from,
+                             final String to) {
+
+        final StringBuilder builder = new StringBuilder();
+
+        boolean found = false;
+        for (String row : marshal.split("\\r?\\n")) {
+            if (found && Objects.equals(row.trim(), to)) {
+                break;
+            }
+            if (found) {
+                final String trim = row.trim();
+                if (!trim.isEmpty()) {
+                    builder.append(trim);
+                    builder.append("\n");
+                }
+            }
+            if (!found && Objects.equals(row.trim(), from)) {
+                found = true;
+            }
+        }
+
+        return builder.toString();
+    }
```

Comment:
This can be rewritten as:
```java
private String subString(final String marshal,
                             final String from,
                             final String to) {

        Pattern regex = Pattern.compile("\\s*" + from + "\\s*(.*)" + to, Pattern.DOTALL);
        Matcher regexMatcher = regex.matcher(marshal);
        if (regexMatcher.find()) {
            return regexMatcher.group(1);
        } else {
            throw new IllegalStateException(String.format("No substring between '%s' and '%s' found",
                                                          from,
                                                          to));
        }
    }
```